src/hotspot/share/memory/filemap.cpp

```diff
@@ -985,8 +985,9 @@ MemRegion FileMapInfo::get_heap_regions_range_with_current_oop_encoding_mode() {
 // open archive objects.
 void FileMapInfo::map_heap_regions_impl() {
   if (!HeapShared::is_heap_object_archiving_allowed()) {
-    log_info(cds)("CDS heap data is being ignored. UseG1GC, "
-                  "UseCompressedOops and UseCompressedClassPointers are required.");
+    log_info(cds)("CDS heap data is being ignored. UseHeapObjectArchiving, "
```

Comment:
constraint checking and warning can be done in arguments.cpp,i.e.

if (UseHeapObjectArchiving) {
if (!G1GC || !UseCompressedOops||...){
warning("....");
UseHeapObjectArchiving=false;
}

}

Reply:
I like your point that filemap.cpp is not the best place for the heap archiving constraint check. The check is only done if the archived data is mapped at runtime, otherwise the check is not necessary (e.g. if CDS is disabled at runtime, -Xshare:off). arguments.cpp doesn't have such context however. 

heapShared.cpp a better a place for such check. There are two reasons to not restructuring the code further for now:

- Introduce new deltas and potential merge conflicts for applying the pre-init patches in the incubator.
- There have been some restructurings in CDS code after JDK 11, partly because of dynamic archiving. The code in this area has been moved around. We can consider refactoring the check when merging into OpenJDK head, which would be needed.

Further thoughts?